spec/suites/map/MapSpec.js

```diff
@@ -611,6 +611,62 @@ describe("Map", function () {
 		});
 	});
 
+	describe("#addHandler", function () {
+		function getHandler(callback = () => {}) {
+			return L.Handler.extend({
+				addHooks: function () {
+					L.DomEvent.on(window, 'click', this.handleClick, this);
+				},
+
+				removeHooks: function () {
+					L.DomEvent.off(window, 'click', this.handleClick, this);
+				},
+
+				handleClick: callback
+			});
+		}
+
+		it("checking enabled method", function () {
+			L.ClickHandler = getHandler();
+			map.addHandler('clickHandler', L.ClickHandler);
+
+			map.clickHandler.enable();
+			expect(map.clickHandler.enabled()).to.eql(true);
+
+			map.clickHandler.disable();
+			expect(map.clickHandler.enabled()).to.eql(false);
+		});
+
+		it("automatically enabled, when has a property named the same as the handler", function () {
+			container = createContainer();
+			map = L.map(container, {clickHandler: true});
```

Comment:
Don't create a new container. Remove the old map and use the same container
```suggestion
			map.remove();
			map = L.map(container, {clickHandler: true});
```

Reply:
Ahh, yeah, makes sense. Fixing.